setup.py

```diff
@@ -0,0 +1,190 @@
+#!/usr/bin/env python
+"""QuTiP: The Quantum Toolbox in Python
+
+QuTiP is open-source software for simulating the dynamics of closed and open
+quantum systems. The QuTiP library depends on the excellent Numpy, Scipy, and
+Cython numerical packages. In addition, graphical output is provided by
+Matplotlib.  QuTiP aims to provide user-friendly and efficient numerical
+simulations of a wide variety of quantum mechanical problems, including those
+with Hamiltonians and/or collapse operators with arbitrary time-dependence,
+commonly found in a wide range of physics applications. QuTiP is freely
+available for use and/or modification on all common platforms. Being free of
+any licensing fees, QuTiP is ideal for exploring quantum mechanics in research
+as well as in the classroom.
+"""
+
+DOCLINES = __doc__.split('\n')
+
+CLASSIFIERS = """\
+Development Status :: 4 - Beta
+Intended Audience :: Science/Research
+License :: OSI Approved :: BSD License
+Programming Language :: Python
+Programming Language :: Python :: 3
+Topic :: Scientific/Engineering
+Operating System :: MacOS
+Operating System :: POSIX
+Operating System :: Unix
+Operating System :: Microsoft :: Windows
+"""
+
+# import statements
+import os
+import sys
+# The following is required to get unit tests up and running.
+# If the user doesn't have, then that's OK, we'll just skip unit tests.
+try:
+    from setuptools import setup, Extension
+    EXTRA_KWARGS = {
+        'setup_require': ['pytest-runner'],
+        'tests_require': ['pytest']
+    }
+except:
+    from distutils.core import setup
+    from distutils.extension import Extension
+    EXTRA_KWARGS = {}
+
+try:
+    import numpy as np
+except ImportError as e:
+    raise ImportError("numpy is required at installation") from e
+
+#from Cython.Build import cythonize
+#from Cython.Distutils import build_ext
+
+# all information about QuTiP goes here
+MAJOR = 0
+MINOR = 0
+MICRO = 1
+ISRELEASED = False
+VERSION = '%d.%d.%d' % (MAJOR, MINOR, MICRO)
+REQUIRES = ['numpy (>=1.12)', 'scipy (>=1.0)', 'qutip (>=4.5)']
+EXTRAS_REQUIRE = {'graphics': ['matplotlib(>=1.2.1)']}
+INSTALL_REQUIRES = ['numpy>=1.12', 'scipy>=1.0', 'qutip>=4.5']
+PACKAGES = ['src/qip', 'src/qip/device', 'src/qip/operations', 'src/qip/compiler',
```

Comment:
You can probably get away with specifying only `src/qutip_qip` here and having a good MANIFEST.in file.

Reply:
`find_packages(where="src")` should do it, where `find_packages` is imported from `setuptools`

Reply:
I have made these changes.

Reply:
See e.g. https://github.com/QAlgebra/qalgebra/blob/master/setup.py (and the corresponding [`MANIFEST.in`](https://github.com/QAlgebra/qalgebra/blob/master/MANIFEST.in) for a project where I use the `src` layout

Reply:
Does it look okay now?